pandas/io/sql.py

```diff
@@ -172,9 +172,15 @@ def _convert_arrays_to_dataframe(
     )
     if dtype_backend == "pyarrow":
         pa = import_optional_dependency("pyarrow")
-        arrays = [
-            ArrowExtensionArray(pa.array(arr, from_pandas=True)) for arr in arrays
-        ]
+        import pyarrow.compute as pc
+
+        result_arrays = []
+        for arr in arrays:
+            pa_array = pa.array(arr, from_pandas=True)
+            if arr.dtype == "string":
+                pa_array = pc.cast(pa_array, pa.string())
```

Comment:
Nit: I think `pa_array.cast(pa.string())` would avoid the extra `pc` import

Reply:
updated